hs_core/views/utils.py

```diff
@@ -924,7 +922,22 @@ def zip_folder(user, res_id, input_coll_path, output_zip_fname, bool_remove_orig
     return output_zip_fname, output_zip_size
 
 
-def unzip_file(user, res_id, zip_with_rel_path, bool_remove_original, overwrite=False):
+class IrodsFile:
+
+    def __init__(self, name, istorage):
+        self._name = name
+        self._istorage = istorage
+
+    @property
+    def name(self):
+        return self._name
+
+    def read(self):
+        return self._istorage.download(self._name).read()
```

Comment:
I question a bit on the usability of this IrodsFile class. Directly calling istorage.download(file_name).read() should be simpler than creating a class to do it in my view.

Reply:
The reason I did this was because it mimics the methods for an uploaded file that are used by the ingestion logic.  This allows the same code to be used for ingestion when a file is uploaded as well as when we unzip a file and ingest the contents.  I'll add a comment explaining this.

Reply:
@hyi I plan a cleanup of these URIs for performance reasons and cleaning this up should wait until I clean up the cause of the URIs.